internal/rbd/rbd_util.go

```diff
@@ -478,6 +479,25 @@ func (rv *rbdVolume) flattenRbdImage(ctx context.Context, cr *util.Credentials,
 	return nil
 }
 
+func (rv *rbdVolume) flatten() error {
+	rbdImage, err := rv.open()
+	if err != nil {
+		return err
+	}
+	defer rbdImage.Close()
+
+	err = rbdImage.Flatten()
+	if err != nil {
+		// go-ceph returns `rbd: ret=22, Invalid argument` if the
```

Comment:
`Invalid argument` is not a unique error for `no parent`. You should probably check with `rbdImage.GetParentInfo()` and see if the name/snapname is stored in the `[]byte` that is passed to the function.

Reply:
this is still a TODO